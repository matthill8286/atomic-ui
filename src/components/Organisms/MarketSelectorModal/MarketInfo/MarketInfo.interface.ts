/* global React */
import { AvailabilityProps } from '@/components/Molecules/Availability/Availability.interface'
import {
  MarketType,
  TranslatedMarketText as TranslatedText,
} from '../MarketSelectorModal.interface'

export interface AddressAndPhoneProps {
  market: MarketType
  showPhoneNumber?: boolean
}

export interface SaveSelectedMarketProps {
  marketId: string
  wcsMarketId: string
  name: string
  availability: AvailabilityProps
}

export interface MarketInfoPropsTranslations {
  buttonSelectDelivery?: TranslatedText
  buttonSelectDeliveryPickup?: TranslatedText
  buttonSelectPickup: TranslatedText
  buttonSelected: TranslatedText
}

export interface MarketInfoMethods {
  closeModal: (e: React.MouseEvent) => void
  saveSelectedMarket: (payload: SaveSelectedMarketProps) => void
}

export interface MarketInfoProps extends MarketInfoPropsTranslations, MarketInfoMethods {
  market: MarketType
  isSelected: boolean
  showBottomBorder?: boolean
  optInConsentComponent?: (market: MarketType) => React.ReactNode | void
}
