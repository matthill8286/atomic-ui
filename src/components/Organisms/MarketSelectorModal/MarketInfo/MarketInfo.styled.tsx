import { Button } from '@/components/Atoms/Button'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'

interface StyledMarketInfoProps {
  showBottomBorder?: boolean
}

export const StyledDistanceBadge = styled.span(
  ({ theme }) => css`
    background: ${theme.color.primary};
    border-radius: ${theme.dimension.borderRadius2};
    color: ${theme.color.white};
    font-size: ${theme.font.size.xxs};
    font-family: ${theme.font.family.default};
    padding: 0 ${theme.spacing.base.xs};
    position: relative;
    top: -3px;
    margin-right: ${theme.spacing.base.xxs};
  `
)

export const StyledMarketInfo = styled.div<StyledMarketInfoProps>(({ showBottomBorder, theme }) =>
  showBottomBorder
    ? css`
        border-bottom: 1px solid ${theme.color.grey2};
        margin-bottom: ${theme.spacing.base.lg};
      `
    : ''
)

export const StyledMarketInfoHeaderRow = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  ${media.md} {
    flex-direction: row;
  }
`

export const StyledMarketInfoRow = styled.div(
  ({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: ${theme.spacing.base.sm} 0;

    ${media.md} {
      flex-direction: row;
      padding: ${theme.spacing.base.sm} 0 ${theme.spacing.base.xl};
    }
  `
)

export const StyledButton = styled(Button)`
  align-self: flex-start;
  max-width: inherit;
  width: 100%;
  margin: ${({ theme }) => theme.spacing.base.sm} 0;

  ${media.md} {
    max-width: 300px;
    flex-direction: row;
    margin: 0;
  }
`

export const StyledLabelParent = styled.div`
  > label {
    height: auto;
    margin-top: ${({ theme }) => theme.spacing.base.sm};
    margin-bottom: ${({ theme }) => theme.spacing.base.md};
  }
  ${media.md} {
    > label {
      margin-top: ${({ theme }) => theme.spacing.base.xs};
      margin-bottom: ${({ theme }) => theme.spacing.base.sm};
    }
  }
`
