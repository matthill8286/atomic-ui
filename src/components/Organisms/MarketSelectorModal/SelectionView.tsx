import React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { CopyText } from '@/components/Atoms/Typography'
import { styled } from '@/styles'
import { OtherCompleteChecked } from '@matthill8286/atomic-icon-library'

export interface SelectionViewProps {
  modalDescription: string
  searchField: React.ReactNode
}

const StyledPickupIcon = styled(Icon)`
  & svg {
    margin-left: auto;
    margin-right: auto;
  }
`

const StyledContainer = styled.div`
  max-width: 390px;
  margin-left: auto;
  margin-right: auto;
`

const StyledCopyText = styled(CopyText)`
  margin-left: auto;
  margin-right: auto;
`

export const SelectionView: React.FC<SelectionViewProps> = ({ modalDescription, searchField }) => {
  return (
    <StyledContainer>
      <StyledPickupIcon width={108} height={108}>
        <OtherCompleteChecked />
      </StyledPickupIcon>
      <StyledCopyText fontSize="md" weight="semibold" tag="p" textAlign="center">
        {modalDescription}
      </StyledCopyText>
      {searchField}
    </StyledContainer>
  )
}
