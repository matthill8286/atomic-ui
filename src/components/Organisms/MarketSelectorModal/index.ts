export * from './MarketSelectorModal'
export * from './MarketSelectorModal.interface'
export * from './MarketInfo/SaturnPin'
