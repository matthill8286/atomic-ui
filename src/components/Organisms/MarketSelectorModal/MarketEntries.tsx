import React from 'react'
import { styled } from '@/styles'
import { MarketInfo } from './MarketInfo/MarketInfo'
import { MarketEntriesProps, MarketType } from './MarketSelectorModal.interface'

const StyledMarketEntries = styled.div`
  margin-top: ${({ theme }) => theme.spacing.base.md};
`

export const MarketEntries: React.FC<MarketEntriesProps> = ({
  buttonSelectDelivery,
  buttonSelectDeliveryPickup,
  buttonSelectPickup,
  buttonSelected,
  closeModal,
  clubMarketId,
  nearbyMarkets,
  saveSelectedMarket,
  selectedMarket,
  optInConsentComponent,
}): React.ReactElement => {
  return (
    <StyledMarketEntries data-test="mms-market-entries">
      {nearbyMarkets &&
        nearbyMarkets.map(
          (market: MarketType, index: number): React.ReactNode => {
            if (typeof market.market_id === 'undefined') {
              return null
            }

            const isSelected =
              (selectedMarket !== null && selectedMarket?.market_id === market.market_id) ||
              clubMarketId === market.market_id

            return (
              <MarketInfo
                buttonSelectDelivery={buttonSelectDelivery}
                buttonSelectDeliveryPickup={buttonSelectDeliveryPickup}
                buttonSelectPickup={buttonSelectPickup}
                buttonSelected={buttonSelected}
                closeModal={closeModal}
                key={market.market_id}
                market={market}
                isSelected={isSelected}
                saveSelectedMarket={saveSelectedMarket}
                showBottomBorder={index < nearbyMarkets.length - 1}
                optInConsentComponent={optInConsentComponent}
              />
            )
          }
        )}
    </StyledMarketEntries>
  )
}
