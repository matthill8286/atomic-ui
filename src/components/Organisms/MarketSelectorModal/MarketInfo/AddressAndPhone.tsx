import React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { CopyText } from '@/components/Atoms/Typography'
import { styled } from '@/styles/styled'
import { OtherPencil, OtherDone } from '@matthill8286/atomic-icon-library'
import { isAlternateTheme } from '@/utils/helper'
import { Address } from '../MarketSelectorModal.interface'
import { AddressAndPhoneProps } from './MarketInfo.interface'
import { MemoIconSePin } from './SaturnPin'

export const buildAddress = (address: Address, lineBreak = false): React.ReactNode => {
  return (
    <>
      {`${address.street}${address.street_number ? ' ' + address.street_number : ''}`}
      {lineBreak ? <br /> : ', '}
      {`${address.zip_code} ${address.city}`}
    </>
  )
}

const StyledAdressRow = styled.div`
  display: flex;
  padding: ${({ theme }) => theme.spacing.base.xxs} 0;
`

const StyledIcon = styled(Icon)`
  position: relative;
  top: 2px;
`

export const AddressAndPhone: React.FC<AddressAndPhoneProps> = ({
  market,
  showPhoneNumber = true,
}: AddressAndPhoneProps): React.ReactElement | null => {
  if (!(market && market.address)) {
    return null
  }
  const color = !isAlternateTheme() ? 'grey3' : undefined
  return (
    <div>
      <StyledAdressRow>
        <StyledIcon width={16} height={16} color={color}>
          {isAlternateTheme() ? <MemoIconSePin /> : <OtherPencil />}
        </StyledIcon>
        <CopyText margin="0 0 0 xs">{buildAddress(market.address, !showPhoneNumber)}</CopyText>
      </StyledAdressRow>
      {showPhoneNumber && (
        <StyledAdressRow>
          <StyledIcon width={16} height={16} color="grey3">
            <OtherDone />
          </StyledIcon>
          {market.address.phone && <CopyText margin="0 0 0 xs">{market.address.phone}</CopyText>}
        </StyledAdressRow>
      )}
    </div>
  )
}
