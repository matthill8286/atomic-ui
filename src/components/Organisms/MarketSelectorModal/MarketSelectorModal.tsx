import React, { useEffect, useState } from 'react'
import { LoadingIndicator } from '@/components/Atoms/LoadingIndicator'
import { Modal } from '@/components/Molecules/Modal'
import { media } from '@/styles/media'
import { styled } from '@/styles/styled'
import { MarketEntries } from './MarketEntries'
import { MarketSelectorModalProps, SaveSelectedMarketProps } from './MarketSelectorModal.interface'
import { SearchField } from './SearchField'
import { SelectionView } from './SelectionView'

const StyledLoadingIndicator = styled(LoadingIndicator)`
  position: relative;
  margin: 0 auto;
`
export const StyledContentWrapper = styled.div<{ noModal?: boolean }>`
  max-width: ${({ noModal }) => (noModal ? '100%' : '400px')};
  margin: 0 auto;
  padding: ${({ theme }) => theme.spacing.base.xs} 0;
  ${media.md} {
    max-width: ${({ noModal }) => (noModal ? '100%' : '650px')};
  }
`

export const MarketSelectorModal: React.FC<MarketSelectorModalProps> = ({
  buttonSelected,
  buttonSelectDelivery,
  buttonSelectDeliveryPickup,
  buttonSelectPickup,
  closeModal,
  clubMarketId,
  consent,
  marketsLoading = false,
  modalDescription,
  modalTitle,
  nearbyMarkets,
  noNearbyStores,
  noResults,
  noResultsMessage,
  saveSelectedMarket,
  searchMarkets,
  searchPlaceholder,
  selectedMarket,
  noModal,
  resetOnSelect,
  hideCloseButton,
  optInConsentComponent,
}) => {
  const [searchQuery, setSearchQuery] = useState('')
  const hideMarketList = resetOnSelect && searchQuery === ''

  const handleSearch = () => {
    if (searchQuery !== '') {
      searchMarkets({ searchQuery })
    }
  }

  useEffect(() => {
    if (selectedMarket?.address) {
      setSearchQuery(selectedMarket.address.city || selectedMarket.address.zip_code)
      handleSearch()
    }
  }, [])

  const saveMarketSelection = (payload: SaveSelectedMarketProps) => {
    saveSelectedMarket(payload)

    if (resetOnSelect) {
      setSearchQuery('')
      resetOnSelect()
    }
  }

  const searchField = (
    <SearchField
      placeholder={searchPlaceholder}
      value={searchQuery}
      onChange={setSearchQuery}
      onSearch={handleSearch}
      consent={consent}
      errorMessage={noResults && noResultsMessage ? noResultsMessage : undefined}
    />
  )

  let content: React.ReactNode = null

  if (marketsLoading) {
    content = <StyledLoadingIndicator color="primary" isVisible size={80} />
  } else if (nearbyMarkets && nearbyMarkets.length === 0) {
    content = (
      <SelectionView
        modalDescription={noResults ? noNearbyStores : modalDescription}
        searchField={searchField}
      />
    )
  } else {
    content = (
      <>
        {searchField}
        {!hideMarketList && (
          <MarketEntries
            buttonSelectPickup={buttonSelectPickup}
            buttonSelectDelivery={buttonSelectDelivery}
            buttonSelectDeliveryPickup={buttonSelectDeliveryPickup}
            buttonSelected={buttonSelected}
            closeModal={closeModal}
            clubMarketId={clubMarketId}
            nearbyMarkets={nearbyMarkets}
            saveSelectedMarket={saveMarketSelection}
            selectedMarket={selectedMarket}
            optInConsentComponent={optInConsentComponent}
          />
        )}
      </>
    )
  }

  return noModal ? (
    <StyledContentWrapper noModal={noModal}>{content}</StyledContentWrapper>
  ) : (
    <Modal title={modalTitle} onClose={closeModal} hideCloseButton={hideCloseButton}>
      <StyledContentWrapper noModal={noModal}>{content}</StyledContentWrapper>
    </Modal>
  )
}
