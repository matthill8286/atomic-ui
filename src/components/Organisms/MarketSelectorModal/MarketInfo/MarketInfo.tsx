import * as React from 'react'
import { Heading } from '@/components/Atoms/Typography'
import { Availability } from '../../../Molecules/Availability'
import { AddressAndPhone } from './AddressAndPhone'
import { MarketInfoProps } from './MarketInfo.interface'
import {
  StyledButton,
  StyledDistanceBadge,
  StyledMarketInfo,
  StyledMarketInfoHeaderRow,
  StyledMarketInfoRow,
  StyledLabelParent,
} from './MarketInfo.styled'
import { StoreState } from '../MarketSelectorModal.interface'

export const MarketInfo: React.FC<MarketInfoProps> = ({
  buttonSelectDelivery,
  buttonSelectDeliveryPickup,
  buttonSelectPickup,
  buttonSelected,
  market,
  saveSelectedMarket,
  isSelected,
  showBottomBorder,
  optInConsentComponent,
}) => {
  if (!market.market_id) {
    return null
  }

  const { availability } = market
  const hasAvailabilityStatus: boolean = market && !!market.availability
  const marketId: string = market.market_id
  const wcsMarketId: string = market.wcs_market_id
  const isUnavailable = hasAvailabilityStatus
    ? market.availability.state === 'NOT_AVAILABLE' || market.state === StoreState.INACTIVE
    : false

  let buttonText = buttonSelectPickup
  if (isSelected) {
    buttonText = buttonSelected
  } else {
    if (buttonSelectDelivery) {
      buttonText = buttonSelectDelivery
    } else if (buttonSelectDeliveryPickup) {
      buttonText = buttonSelectDeliveryPickup
    }
  }

  const ActionButton = (
    <StyledButton
      disabled={isUnavailable}
      onClick={() => {
        saveSelectedMarket({
          marketId,
          wcsMarketId,
          name: market.name,
          availability: market.availability,
        })
      }}
      actionType={isSelected ? 'secondary' : 'primary'}
      fullWidth>
      {buttonText}
    </StyledButton>
  )

  return (
    <StyledMarketInfo showBottomBorder={showBottomBorder}>
      <StyledMarketInfoHeaderRow>
        <div>
          {market.distance && <StyledDistanceBadge>{market.distance}</StyledDistanceBadge>}
          {market.name && (
            <Heading scale="level-4" tag="span">
              {market.name}
            </Heading>
          )}
        </div>
        {hasAvailabilityStatus && (
          <Availability text={availability.text} state={availability.state} />
        )}
      </StyledMarketInfoHeaderRow>
      <StyledMarketInfoRow>
        <AddressAndPhone market={market} showPhoneNumber={false} />
        {optInConsentComponent !== undefined ? (
          <StyledLabelParent>
            {optInConsentComponent(market)}
            {ActionButton}
          </StyledLabelParent>
        ) : (
          ActionButton
        )}
      </StyledMarketInfoRow>
    </StyledMarketInfo>
  )
}
