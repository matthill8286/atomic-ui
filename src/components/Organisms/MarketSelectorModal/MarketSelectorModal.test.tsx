import React from 'react'
import { LoadingIndicator } from '@/components/Atoms/LoadingIndicator'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { MarketEntries } from './MarketEntries'
import { MarketInfo } from './MarketInfo'
import { MarketSelectorModal, StyledContentWrapper } from './MarketSelectorModal'
import { MarketSelectorModalProps } from './MarketSelectorModal.interface'
import { market, SharedProps } from './MarketSelectorModal.mock'
import { SelectionView, SelectionViewProps } from './SelectionView'

const DefaultProps: MarketSelectorModalProps = {
  ...SharedProps,
  closeModal: jest.fn(),
  saveSelectedMarket: jest.fn(),
}

describe('<MarketSelectorModal />', () => {
  const props: MarketSelectorModalProps = {
    ...DefaultProps,
    clubMarketId: '',
    nearbyMarkets: [],
    selectedMarket: market,
  }

  it('renders correct', () => {
    const tree = renderWithTheme(<MarketSelectorModal {...DefaultProps} />)
    expect(tree).toMatchSnapshot()
  })

  test('Should render withput crashing', () => {
    const component = mountWithTheme(<MarketSelectorModal {...props} />)
    expect(component.find(MarketSelectorModal)).toHaveLength(1)
  })

  test('Renders the container', () => {
    const component = mountWithTheme(<MarketSelectorModal {...props} />)
    expect(component.find(StyledContentWrapper)).toHaveLength(1)
  })

  describe('marketsLoading property:', () => {
    describe('<SelectionView />', () => {
      test('Should render if !marketsLoading', () => {
        const component = mountWithTheme(<MarketSelectorModal {...props} marketsLoading={false} />)
        expect(component.find(SelectionView)).toHaveLength(1)
      })

      test('Should not render if marketsLoading', () => {
        const component = mountWithTheme(<MarketSelectorModal {...props} marketsLoading />)
        expect(component.find(SelectionView)).toHaveLength(0)
      })
    })

    describe('<LoadingIndicator />', () => {
      test('Should render if marketsLoading', () => {
        const component = mountWithTheme(<MarketSelectorModal {...props} marketsLoading />)
        expect(component.find(LoadingIndicator)).toHaveLength(1)
      })

      test('Should not render if !marketsLoading', () => {
        const component = mountWithTheme(<MarketSelectorModal {...props} marketsLoading={false} />)
        expect(component.find(LoadingIndicator)).toHaveLength(0)
      })
    })

    describe('<MarketEntries />', () => {
      test('Should render if !marketsLoading and nearbyMarkets.length > 1', () => {
        const component = mountWithTheme(
          <MarketSelectorModal {...props} marketsLoading={false} nearbyMarkets={[{ ...market }]} />
        )
        expect(component.find(MarketEntries)).toHaveLength(1)
      })
      test('Should not render if marketsLoading', () => {
        const component = mountWithTheme(
          <MarketSelectorModal {...props} marketsLoading nearbyMarkets={[{ ...market }]} />
        )
        expect(component.find(MarketEntries)).toHaveLength(0)
      })
    })
  })

  describe('SelectionView():', () => {
    const ssmProps: SelectionViewProps = {
      searchField: <input />,
      modalDescription: 'No results',
    }

    test('Should render without crashing', () => {
      const component = mountWithTheme(<SelectionView {...ssmProps} />)
      expect(component.find(SelectionView)).toHaveLength(1)
      expect(component.find('input')).toHaveLength(1)
    })
  })

  describe('MarketEntries():', () => {
    test('Should render without crashing', () => {
      const component = mountWithTheme(<MarketEntries {...props} />)
      expect(component).toHaveLength(1)
    })

    test('Should not render any MarketInfo if no nearbyMarkets', () => {
      const component = mountWithTheme(<MarketEntries {...props} />)
      expect(component.find(MarketInfo)).toHaveLength(0)
    })

    test('Should render MarketInfo if nearbyMarkets', () => {
      const component = mountWithTheme(<MarketEntries {...props} nearbyMarkets={[{ ...market }]} />)
      expect(component.find(MarketInfo)).toHaveLength(1)
    })
  })
})
