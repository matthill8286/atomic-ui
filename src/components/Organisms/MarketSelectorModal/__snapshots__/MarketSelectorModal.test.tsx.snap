// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarketSelectorModal /> renders correct 1`] = `
.c5 {
  margin: 0;
  font-size: 26px;
  line-height: 28px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  text-align: center;
}

.c15 {
  color: #666666;
  font-size: 18px;
  line-height: 28px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
}

.c21 {
  color: #666666;
  font-size: 16px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 svg {
  width: 30px;
  height: 30px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #666666;
}

.c7 svg g,
.c7 svg p {
  fill: #666666;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 svg {
  width: 108px;
  height: 108px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: currentColor;
}

.c13 svg g,
.c13 svg p {
  fill: currentColor;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c27 svg {
  width: 20px;
  height: 20px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #CCCCCC;
}

.c27 svg g,
.c27 svg p {
  fill: #CCCCCC;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  border: 2px solid #00C8DC;
  background: #FFFFFF;
  color: #00C8DC;
}

.c25 .c6 {
  padding: 0 0px;
}

.c25 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c25:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c25 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c26 {
  min-width: 20px;
  height: 20px;
  padding: 0 8px;
}

.c0 {
  z-index: 201;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  position: fixed;
  overflow-y: auto;
}

.c1 {
  position: fixed;
  position: -webkit-sticky;
  position: sticky;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.7);
}

.c2 {
  background: #FFFFFF;
  border-radius: 0;
  box-sizing: border-box;
  z-index: 101;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  overflow-y: hidden;
  position: absolute;
  top: 0;
}

.c10 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.c10 img {
  max-width: 100%;
}

.c9 {
  padding: 0 16px 16px 16px;
  max-width: 100%;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 0;
  z-index: 1;
  outline: none;
  background: none;
  border: none;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  vertical-align: top;
  width: 100%;
}

.c18 .c6 {
  position: absolute;
  right: 0;
  top: 28px;
  outline: none;
}

.c18 .c6:focus svg {
  fill: #000000;
}

.c23 {
  border: 0;
  box-sizing: content-box;
  display: block;
  font-size: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  margin: 24px 0 0 0;
  box-shadow: 0 3px 4px 0 rgba(0,0,0,0.07),0 3px 3px -2px rgba(0,0,0,0.06),0 1px 8px 0 rgba(0,0,0,0.10);
  border-radius: 12px;
  background-color: #FFFFFF;
  min-width: 0;
  width: 100%;
  padding: 8px 16px;
  line-height: 40px;
  outline: none;
  position: relative;
  -webkit-tap-highlight-color: transparent;
}

.c23::-webkit-input-placeholder {
  opacity: 0;
}

.c23::-moz-placeholder {
  opacity: 0;
}

.c23:-ms-input-placeholder {
  opacity: 0;
}

.c23::placeholder {
  opacity: 0;
}

.c23:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c23:focus::-moz-placeholder {
  opacity: 1;
}

.c23:focus:-ms-input-placeholder {
  opacity: 1;
}

.c23:focus::placeholder {
  opacity: 1;
}

.c24 {
  position: absolute;
  top: 16px;
  right: 16px;
  height: 100%;
}

.c24 > button {
  background: none;
  border: none;
  margin: 0;
  padding: 0;
  top: 28px;
  outline: none;
}

.c24 > button:focus svg {
  fill: #000000;
}

.c20 {
  margin-right: 32px;
}

.c22 {
  position: absolute;
  margin-top: 4px;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: color 0.25s cubic-bezier(0,0,0.2,1) 0ms,top 0.25s cubic-bezier(0,0,0.2,1) 0ms;
  transition: color 0.25s cubic-bezier(0,0,0.2,1) 0ms,top 0.25s cubic-bezier(0,0,0.2,1) 0ms;
  top: 24px;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
}

.c19 input::-webkit-input-placeholder {
  opacity: 1;
}

.c19 input::-moz-placeholder {
  opacity: 1;
}

.c19 input:-ms-input-placeholder {
  opacity: 1;
}

.c19 input::placeholder {
  opacity: 1;
}

.c19 .c6 {
  cursor: pointer;
}

.c17 {
  position: relative;
  min-height: auto;
}

.c14 svg {
  margin-left: auto;
  margin-right: auto;
}

.c12 {
  max-width: 390px;
  margin-left: auto;
  margin-right: auto;
}

.c16 {
  margin-left: auto;
  margin-right: auto;
}

.c11 {
  max-width: 400px;
  margin: 0 auto;
  padding: 8px 0;
}

@media (min-width:512px) {
  .c1 {
    overflow: auto;
    -webkit-overflow-scrolling: touch;
  }
}

@media (min-width:512px) {
  .c2 {
    top: 48px;
    -webkit-transform: translate(-50%);
    -ms-transform: translate(-50%);
    transform: translate(-50%);
    margin-bottom: 64px;
    left: 50%;
    display: block;
    padding: 32px 24px;
    border-radius: 4px;
    height: auto;
    max-width: 756px;
  }
}

@media (min-width:512px) {
  .c9 {
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
    padding: 0;
  }
}

@media (min-width:512px) {
  .c3 {
    padding: 0 0 16px 0;
  }
}

@media (min-width:752px) {
  .c11 {
    max-width: 650px;
  }
}

<div
  className="c0"
  id="saiyan-styled-modal-wrapper"
>
  <div
    className="c1"
    onClick={[Function]}
  />
  <div
    className="c2"
    onClick={[Function]}
    size="lg"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      />
      <div>
        <h1
          className="c5"
          fontFamily="default"
          fontSize="xl"
          spacing="base"
        >
          Choose a market
        </h1>
      </div>
      <div
        className="c6 c7 c8"
        color="grey5"
        height={30}
        onClick={[Function]}
        width={30}
      >
        <svg
          height={32}
          viewBox="0 0 24 24"
          width={32}
        />
      </div>
    </div>
    <div
      className="c9"
      id="saiyan-styled-modal-desktop-wrapper"
    >
      <div
        className="c10"
      >
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            <div
              className="c6 c13 c14"
              height={108}
              width={108}
            >
              <svg
                height={20}
                width={20}
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <defs>
                  <path
                    d="M10 0c5.514 0 10 4.486 10 10s-4.486 10-10 10S0 15.514 0 10 4.486 0 10 0zm0 1.622C5.38 1.622 1.622 5.38 1.622 10S5.38 18.378 10 18.378 18.378 14.62 18.378 10 14.62 1.622 10 1.622zm3.848 5.06a.81.81 0 111.116 1.177l-5.757 5.46a.81.81 0 01-1.14-.024l-3.026-3.107a.81.81 0 111.161-1.132l2.47 2.535z"
                    id="a"
                  />
                </defs>
                <g
                  fill="none"
                  fillRule="evenodd"
                >
                  <mask
                    fill="#fff"
                    id="b"
                  >
                    <use
                      xlinkHref="#a"
                    />
                  </mask>
                  <use
                    fill="#000"
                    fillRule="nonzero"
                    xlinkHref="#a"
                  />
                  <g
                    fill="#999"
                    mask="url(#b)"
                  >
                    <path
                      d="M0 0h20v20H0z"
                    />
                  </g>
                </g>
              </svg>
            </div>
            <p
              className="c15 c16"
              color="grey5"
              fontSize="md"
            >
              Please enter your postal code or city, to view markets in your area.
            </p>
            <div
              className="c17"
              data-test="mms-market-search-field"
            >
              <div
                className="c18 c19"
              >
                <div
                  className="c20"
                >
                  <label
                    className="c21 c22"
                    color="grey5"
                    fontSize="sm"
                  >
                    
                  </label>
                  <input
                    aria-invalid={false}
                    aria-required={false}
                    autoFocus={true}
                    className="c23"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    placeholder="Search..."
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="c24"
                >
                  <button
                    className="c25 c26"
                    disabled={false}
                    height={20}
                    onClick={[Function]}
                    size="md"
                    type="button"
                    width={20}
                  >
                    <span
                      className="c6 c27"
                      color="grey3"
                      height={20}
                      width={20}
                    >
                      <svg
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M20.85 19.44l-4-4a2.44 2.44 0 00-.43-.35l-1-.69A7 7 0 1010 17a7 7 0 004.37-1.53l.75 1a2.6 2.6 0 00.3.36l4 4a.5.5 0 00.71 0l.7-.7a.5.5 0 00.02-.69zM10 15a5 5 0 115-5 5 5 0 01-5 5z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
